### PR TITLE
Company subscription  image issue fix

### PR DESCRIPTION
**Pull request by manojava-gk**

### CODE REVIEW

**Comment by manojava-gk** [comment]
could not get rid of any type as it is from Promise function. Compilation failed by saying unknown[] is not assignable to a type

File: cx-portal/src/services/CommonService.ts [hunk 1]
```diff
@@ -51,18 +51,28 @@ const fetchLeadPictureImage = (data: any[]) => {
...
+                    })
+                  } else {
+                    resolve({
+                      ...app,
+                      image: {
+                        src: this.result,
+                        alt: app.title,
+                      },
+                    })
+                  }
```
Comment (oyo) [inline cx-portal/src/services/CommonService.ts:75]: a cleaner way to write this:
`reader.onload = () => resolve(doNotPickDefaultType ? {...obj_from_else...} : {...obj_from_if...})`

File: cx-portal/src/components/pages/Organization/index.tsx [hunk 2]
```diff
@@ -44,8 +47,7 @@ export default function Organization() {
     isLoading,
   } = useFetchSubscriptionStatusQuery()
   const { data } = useFetchActiveAppsQuery()
-  const appSubscribedData =
-    data && subscriptionStatus && appToStatus(data, subscriptionStatus)
+  const [appSubscribedData, setAppSubscribedData] = useState<any>([])
```
Comment (oyo) [inline cx-portal/src/components/pages/Organization/index.tsx:50]: avoid any - define a type or interface for `appSubscribedData`

File: cx-portal/src/components/pages/Organization/index.tsx [hunk 3]
```diff
@@ -58,18 +60,44 @@ export default function Organization() {
...
-      ),
-    ]) || []
+    appSubscribedData?.map(
+      (app: {
+        image: ImageType | undefined
+        id: string | undefined
+        name: any
+        provider: string
+        status: SubscriptionStatus | undefined
+      }) => [
```
Comment (oyo) [inline cx-portal/src/components/pages/Organization/index.tsx:89]: define a type or interface for complex objects

File: cx-portal/src/components/pages/Organization/index.tsx [hunk 4]
```diff
@@ -58,18 +60,44 @@ export default function Organization() {
...
+            const status = subscriptionStatus?.find(
+              (e) => e.appId === app.id
+            )?.offerSubscriptionStatus
+            return { ...app, status }
+          })
+          .filter((e) => e.status)
+        setAppSubscribedData(newApps)
+      })
+    }
+  }, [data, subscriptionStatus])
```
Comment (oyo) [inline cx-portal/src/components/pages/Organization/index.tsx:79]: this is very difficult to read - better something like (schematic)

`Promise.all(newPromies).then((result) =>
  setAppSubscribedData(
    result
      .flat()
      .map((app: SomeTypeOrInterface) => ({
        ...app,
        subscriptionStatus?.find(...)
      }))
  )
)`

**Review by oyo — CHANGES_REQUESTED** [review]
code difficult to read and understand see suggestions to simplify

File: cx-portal/src/components/pages/Organization/index.tsx (hunk 2 above)
Comment (oyo) [reply cx-portal/src/components/pages/Organization/index.tsx]: @manojava-gk just saw your comment on the `any` - I'm going to check out the branch to see

File: cx-portal/src/services/CommonService.ts [hunk 5]
```diff
@@ -51,18 +51,28 @@ const fetchLeadPictureImage = (data: any[]) => {
...
-                    description: getDescription(app),
-                    price: getPrice(app),
-                    onClick: onClick(app),
-                    image: {
-                      src: this.result,
-                      alt: app.title,
-                    },
-                  })
+                  resolve(
+                    !doNotPickDefaultType
```
Comment (oyo) [inline cx-portal/src/services/CommonService.ts:55]: instead of
`!condition ? a : b`
better write
`condition ? b : a`

**Review by oyo — CHANGES_REQUESTED** [review]
one more remark

File: cx-portal/src/services/CommonService.ts (hunk 5 above)
Comment (manojava-gk) [reply cx-portal/src/services/CommonService.ts:55]: As value itself has default value as false, I wrote like this. Will update it

**Review by nidhigarg-bmw — DISMISSED** [review]

File: cx-portal/src/components/pages/Organization/index.tsx [hunk 6]
```diff
@@ -58,6 +60,25 @@ export default function Organization() {
     dispatch(show(OVERLAYS.APP, id, t('content.organization.company.title')))
   }
 
+  useEffect(() => {
+    if (data && subscriptionStatus) {
+      const newPromies = CommonService.fetchLeadPictureImage(data, true)
```
Comment (oyo) [inline cx-portal/src/components/pages/Organization/index.tsx:65]: no `Promise.all` for image loading - use the new Image component (see https://jira.catena-x.net/browse/CPLP-2284 )

**Review by oyo — CHANGES_REQUESTED** [review]
Image loading is using the bad design pattern with `Promise.all` - please change. Have a look at ´AppOverviewNew` page how to use

**Comment by manojava-gk** [comment]
Will create a new PR using new Image component

**Pull request closed without merging**